src/ValueParsers/ParseException.php

```diff
@@ -23,11 +23,11 @@ class ParseException extends RuntimeException {
 	private $rawValue;
 
 	/**
-	 * @param string $message        A plain english message describing the error
+	 * @param string $message A plain english message describing the error
 	 * @param string|null $rawValue The raw value that failed to be parsed.
 	 * @param string|null $expectedFormat An identifier for the format the raw value did not match
 	 */
-	public function __construct( string $message, string $rawValue = null, string $expectedFormat = null ) {
+	public function __construct( string $message, ?string $rawValue = null, ?string $expectedFormat = null ) {
```

Comment:
Huh. Good to know. https://php.watch/versions/8.4/implicitly-marking-parameter-type-nullable-deprecated